packages/theia-integration/src/browser/theia-jsonrpc-glsp-client.ts

```diff
@@ -45,12 +53,30 @@ export class TheiaJsonrpcGLSPClient extends BaseJsonrpcGLSPClient {
         }
         return super.checkConnectionState();
     }
+
+    protected override doCreateConnection(): Promise<MessageConnection> {
+        if (this.webSocketPort) {
+            const websocket = new WebSocket(`ws://localhost:${this.webSocketPort}/${this.id}`);
+            return new Promise(resolve => {
```

Comment:
You could reuse the `listen` utility function from `@eclipse-glsp/protocol` here